lib/core/Queue.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+class Queue {
+  constructor() {
+    this.list = [];
+  }
+
+  enqueue(data) {
+    this.list.push(data);
+  }
+
+  dequeue() {
+    if (this.isEmpty()) {
+      return null;
+    }
+    return this.list.shift();
```

Comment:
just beware that this is O(n) operation. If there are multiple connections in pause mode and all of them start draining/dequeuing at once (and if there were multiple messages in it), then this could be the future chokepoint for a moment.
Not a very big deal.
Just FYI...

Reply:
Noted for now keeping it simple